R/clean_names.R

```diff
@@ -81,14 +82,19 @@ clean_names.default <- function(dat, ...) {
   if (is.null(names(dat))) {
     dimnames(dat) <- lapply(dimnames(dat), make_clean_names, ...)
   } else {
+    if (set_labels){
+      old_names <- names(dat)
+      for (i in seq_along(old_names)) attr(dat[[i]], "label") <- old_names[[i]]
+    }
     names(dat) <- make_clean_names(names(dat), ...)
+    
   }
   dat
 }
 
 #' @rdname clean_names
 #' @export
-clean_names.sf <- function(dat, ...) {
+clean_names.sf <- function(dat, ..., set_labels = FALSE) {
```

Comment:
@sfirke, I don't know enough about `sf` objects.  Do you know if adding labels may cause any issues?  (In other words, should `set_labels` apply to this function?)

Reply:
As far as I know, it shouldn't affect at all. All operations done in the columns that are not the geometry-column are the same as done in a data.frame or tibble. I work often with sf objects, and I've never found a case where labels might suppose a problem.

Reply:
I will defer to @jospueyo , I have never used sf objects and that sounds reasonable.